tap4j/src/main/java/org/tap4j/model/Patterns.java

```diff
@@ -58,7 +58,7 @@ private Patterns() { // hidden as this is a final utility class
     /**
      * TAP Test Result Regex.
      */
-    static final String REGEX_TEST_RESULT = "((\\s|\\t)*)?(ok|not ok)\\s*(\\d*)\\s*([^#]*)?\\s*"
+    static final String REGEX_TEST_RESULT = "(\\s*)(ok|not ok)\\s*(\\d*)\\s*([^#]*)?\\s*"
```

Comment:
Any reason why we should remove tabs here too?

Reply:
The thing is `\s` includes `\t`, so it's redundant to use alternating matching between them. See https://docs.oracle.com/javase/tutorial/essential/regex/pre_char_classes.html

Reply:
And this minor issue is present in other regexes too, but I didn't fix it, because not sure if wanted.

Reply:
Just passed midnight here, so the new day barely started and I can proudly say Today-I-Learned.

Had no idea they were redundant. I've been uaing both in Perl, shell, java... might have a few places to fix besides tap4j.

This change is definitely wanted. But we can fix in another issue if that's easier.

Thanks!!!

Reply:
Also, `*` takes exactly zero or more elements, so you can omit redundant `?`.

Reply:
Right. I think you are talking about this part:

`([^#]*)?`

I thought that it would resolve `[^#]*`as zero or more of `[^#]` in a group `()?`. And so the group was receiving the `?`, and thus would appear zero or at most once?

Reply:
Aahh! Sure. You fixed that one already, but I agree for the remaining regular expressions. Gonna start thinking a bit more when writing regexes now :)